js/src/common/utils/extract.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * The `extract` utility deletes a property from an object and returns its
+ * value.
+ *
+ * @param object The object that owns the property
+ * @param property The name of the property to extract
+ * @return The value of the property
+ */
+export default function extract<T, K extends keyof T>(object: T, property: K): T[K] {
```

Comment:
Generics... crazy stuff. :scream:

Reply:
Yep! This was actually new to me as well when I wrote this. It basically means that when you do `extract({ id: 2 }, 'id')`:
- `T` is the type of `{ id: number }`
- `K` is the name of a property of T, so `'id'`
- `T[K]` is the type of the property `T[K]`, or `object[property]`, in this case `number`

Hopefully this makes sense & I explained it correctly.